### DISCUSSION TITLE
Update Consumer config

### DISCUSSION THREAD

**Discussion by MrAolen** [spring-projects/spring-kafka#1844] [Q&A] [answered]
Hi,

Is it possible to update consumer config after their instantiation ? In particular, I would like to add interceptor to the config. 

Thanks in advance.

**Comment by garyrussell** [comment, marked as answer]
Yes, since version 2.7 you can modify the consumer factory properties.

https://github.com/spring-projects/spring-kafka/blob/8bcb0529e2f0875b3499ae74aaa46a21200f47e3/spring-kafka/src/main/java/org/springframework/kafka/core/ConsumerFactory.java#L207-L214

Since version 2.3, you can override/supplement the consumer factory properties via the container properties.

https://github.com/spring-projects/spring-kafka/blob/8bcb0529e2f0875b3499ae74aaa46a21200f47e3/spring-kafka/src/main/java/org/springframework/kafka/listener/ConsumerProperties.java#L319-L348

In either case, you have to restart the container(s) for it to take effect.

**Reply by MrAolen** [reply]
Works like a charm. Thanks !

**Reply by KhaiNguyenDuc** [reply]
Thanks, It's work
